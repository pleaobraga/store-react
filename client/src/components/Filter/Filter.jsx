import React from 'react'
import PropTypes from 'prop-types'
import './Filter.scss'

export const Filter = ({ value, onChange, placeholder }) => {
  return (
    <div className="filter">
      <i key="search" className="material-icons filter__icon">
        search
      </i>
      <input
        className="filter__input"
        placeholder={placeholder ? placeholder : 'Search'}
        value={value}
        onChange={onChange}
      />
    </div>
  )
}

Filter.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  placeholder: PropTypes.string
}

export default React.memo(Filter)
