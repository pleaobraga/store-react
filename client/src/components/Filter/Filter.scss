@import '../../styles/colors';

.filter {
  display: flex;
  align-items: center;

  width: 100%;
  padding: 2px;

  border: 1px solid $black;
  border-radius: 10px;

  &__icon {
    font-size: 2rem;
    padding-left: 1rem;
  }

  &__input {
    width: 100%;
    padding-left: 1rem;
    border: none;

    &:focus {
      outline: none;
    }
  }
}
