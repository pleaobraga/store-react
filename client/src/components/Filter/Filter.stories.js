import React from 'react'
import { storiesOf } from '@storybook/react'
import { Filter } from './Filter'

export default storiesOf('Components | Filter', module).add(
  'default',
  () => <Filter value={'Test'} onChange={() => {}} />,
  {
    info: {
      inline: true,
      header: false,
      propTables: [Filter]
    }
  }
)
