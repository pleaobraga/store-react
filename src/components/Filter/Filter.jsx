import React from 'react'
import PropTypes from 'prop-types'
import './Filter.scss'

export const Filter = ({ value, onChange }) => {
  return (
    <div className="filter">
      <i key="search" className="material-icons filter__icon">
        search
      </i>
      <input className="filter__input" value={value} onChange={onChange} />
    </div>
  )
}

Filter.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func
}

export default React.memo(Filter)
