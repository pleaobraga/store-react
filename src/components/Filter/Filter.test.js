import React from 'react'
import { shallow } from 'enzyme'
import Filter from './Filter'

describe('Filter', () => {
  const filter = shallow(<Filter value="test" onChange={() => {}} />)

  it('render proprely', () => {
    expect(filter).toMatchSnapshot()
  })
})
