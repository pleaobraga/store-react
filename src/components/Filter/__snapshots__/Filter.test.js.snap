// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter render proprely 1`] = `
<div
  className="filter"
>
  <i
    className="material-icons filter__icon"
    key="search"
  >
    search
  </i>
  <input
    className="filter__input"
    onChange={[Function]}
    value="test"
  />
</div>
`;
